File: app/src/main/java/com/jobhub/domain/Recruitment.java
package com.jobhub.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Comment;

import java.time.LocalDateTime;
import java.util.Objects;


@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
@Entity
@Table(name = "recruitment",
        uniqueConstraints={
                @UniqueConstraint(columnNames = {"signed_hash"}, name = "UK_RECRUITMENT_SIGNED_HASH"),
        }
)
public class Recruitment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    @Comment("공고 URL")
    private String url;

    @Column(nullable = false)
    @Comment("공고 제공처")
    private String provider;

    @Column(nullable = false)
    @Comment("공고 제목")
    private String title;

    private String department;

    @Column(name = "company_name", nullable = false)
    private String companyName;

    @Column(name = "company_address")
    private String companyAddress;

    @Column(name = "start_date")
    private LocalDateTime startDate;

    @Column(name = "end_date")
    private LocalDateTime endDate;

    @Column(name = "signed_hash", nullable = false)
    @Comment("해시")
    private String signedHash;

    @Builder
    public Recruitment(String url,
                       String provider,
                       String title,
                       String department,
                       String companyName,
                       String companyAddress,
                       LocalDateTime startDate,
                       LocalDateTime endDate,
                       String signedHash) {
        Objects.requireNonNull(url);
        Objects.requireNonNull(title);
        Objects.requireNonNull(provider);
        Objects.requireNonNull(companyName);
        Objects.requireNonNull(signedHash);
        this.url = url;
        this.provider = provider;
        this.title = title;
        this.department = department;
        this.companyName = companyName;
        this.companyAddress = companyAddress;
        this.startDate = startDate;
        this.endDate = endDate;
        this.signedHash = signedHash;
    }

    public void change(Recruitment newRecruitment) {
        this.title = newRecruitment.title;
        this.startDate = newRecruitment.startDate;
        this.endDate = newRecruitment.endDate;
    }
}

File: crawling/src/main/python/Recruitment.py
class Recruitment:
    def __init__(self, id, url, provider, title, department, career
                 , company_name, company_address, start_date, end_date, signed_hash):
        self.id = id
        self.url = url
        self.provider = provider
        self.title = title
        self.department = department
        self.career = career
        self.companyName = company_name
        self.companyAddress = company_address
        self.startDate = start_date
        self.endDate = end_date
        self.signedHash = signed_hash

    def to_dict(self):
        return {
            'id': self.id,
            'url': self.url,
            'provider': self.provider,
            'title': self.title,
            'department': self.department,
            'career': self.career,
            'companyName': self.companyName,
            'companyAddress': self.companyAddress,
            'startDate': self.startDate,
            'endDate': self.endDate,
            'signedHash': self.signedHash
        }
